### PR TITLE
Import projects route and tests

### PR DESCRIPTION
**Pull request by jsvoo**

### CODE REVIEW

File: classes/Project/__tests__/exists_unit.test.mjs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+ import ImportProject from '../ImportProject.mjs';
+
+describe('ImportProject Class #importTests', () => {
+  it('should have a constructor', () => {
+    expect(ImportProject.prototype.constructor).toBeInstanceOf(Function);
+  });
```
Comment (thehabes) [inline classes/Project/__tests__/exists_unit.test.mjs:6]: remove superfluous `;` and all others like it in this file.

File: classes/Project/__tests__/functionality_unit.test.mjs [hunk 2]
```diff
@@ -0,0 +1,110 @@
+ import { jest } from '@jest/globals';
+import ImportProject from '../ImportProject.mjs';
+
+describe('ImportProject.fetchManifest #importTests', () => {
+  beforeEach(() => {
+    global.fetch = jest.fn();
```
Comment (thehabes) [inline classes/Project/__tests__/functionality_unit.test.mjs:6]: remove superfluous `;` and all others like it in this file.

File: classes/Project/ImportProject.mjs [hunk 3]
```diff
@@ -6,9 +6,8 @@ export default class ImportProject {
     this.data = data
   }
 
-  static async fetchManifest(manifestId) {
-    const url = `https://t-pen.org/TPEN/project/${manifestId}`
-    return fetch(url)
+  static async fetchManifest(url) {
+     return fetch(url)
```
Comment (thehabes) [inline classes/Project/ImportProject.mjs:10]: Gotta catch() somewhere

File: classes/Project/ImportProject.mjs [hunk 4]
```diff
@@ -56,19 +55,15 @@ export default class ImportProject {
     return layers
   }
 
-  static async fromManifest(manifestId) {
+  static async fromManifestURL(manifestId) {
```
Comment (thehabes) [inline classes/Project/ImportProject.mjs:58]: Gotta catch() somewhere

**Review by thehabes — CHANGES_REQUESTED** [review]
Test the endpoint using the following JSON bodies and see if you can make the responses a bit more RESTful.
```JSON
{
   "url": "https://tinydev.rerum.io/plain.txt"
}
```

```JSON
{
   "url": "https://fake.com/id/123123"
}
```

```JSON
{
   "url": 123
}
```
image

The `@id` of the layer should not be the `@id` of the 'canvas', as seen here in this success response

```JSON
{
   "title":"Kerpen, Kath. Pfarrei St. Martinus Codex 0009 project",
   "@context":"http://t-pen.org/3/context.json",
   "manifest":"https://store.rerum.io/v1/id/667d8064a149c276746683aa",
   "layers":[
      {
         "@id":"https://t-pen.org/TPEN/canvas/105420",
         "@type":"Layer",
         "pages":[
            
         ]
      }
   ],
   "@type":"Project",
   "_id":"6684375379b1439e8e6389d9"
}
```

The line of code is in ImportProject.processLayerFromCanvas()
```
layer["@id"] = canvas["@id"]
```
The layer `@id` will be the store.rerum.io/v1/id/123123123 of some AnnotationCollection in RERUM instead, I believe.

File: project/index.mjs [hunk 5]
```diff
@@ -243,15 +305,60 @@ router.get('/:id', async (req, res, next) => {
...
+      .json({
+        message: `Import from ${createFrom} is not available. Create from URL instead`
+      })
   }
 })
 
-router.all('/', (req, res, next) => {
-  utils.respondWithError(res, 405, 'Improper request method, please use GET.')
+router.all("/", (req, res, next) => {
+  utils.respondWithError(res, 405, "Improper request method, please use GET.")
```
Comment (thehabes) [inline project/index.mjs:361]: please use POST

**Review by cubap — APPROVED** [review]

**Pull request merged**